### PR TITLE
Fixed geotrace and geoshape widgets in read-only mode

### LINKED ISSUE #6096: Save button is active in Geo widgets with readonly option
#### ODK Collect version
the master version 4f54bcce2d548e9869b2ec6fbf9e674d37e77fa1, the store version 2024.1.3

#### Android version 
10, 11, 14

#### Device used
Redmi 9T, Samsung M12, Pixel 7a

#### Problem description
Save button is active in Geo widgets with readonly option, due to this it is possible to change answers in GeoTrace and GeoShape questions by moving points on map and then pressing save button. Additionally pressing save button in Geo questions with readonly option, results in changing button from "View ...." to "View or change ...."

#### Steps to reproduce the problem
1. Open Geo read-only form
2. Go to GeoTrace question
3. Press "View line" button
4. Move points
5. Press save button
#### Expected behavior
Save button should be inactive in Geo widgets with readonly option

#### Other information 

https://github.com/getodk/collect/assets/162433284/d5764878-8a22-4676-b0a4-8d3bb17a2599

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #6096

#### Why is this the best possible solution? Were any other approaches considered?
Apart from disabling the save icon in read-only mode, I've decided to make lines and polygons not draggable: https://github.com/getodk/collect/pull/6102/commits/6b87ee2f622ecf1264ff324759cc4cafbb110261 it was a problem with the disabled saved button but was confusing so I think we should introduce both improvements,

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
This should disable the save button in read-only mode and also make lines and polygons not draggable. Nothing else should be affected but we need to test the maps as a whole to make sure that everything is fine.

#### Do we need any specific form for testing your changes? If so, please attach one.
The form is attached to the issue.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew connectedAndroidTest` (or `./gradlew testLab`) and confirmed all checks still pass
- [x] added a comment above any new strings describing it for translators
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: geo/src/test/java/org/odk/collect/geo/geopoly/GeoPolyActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    @Test
+    fun buttonsShouldBeEnabledInEditableMode() {
+        val polyline = ArrayList<MapPoint>()
+        polyline.add(MapPoint(1.0, 2.0, 3.0, 4.0))
+        val intent =
+            Intent(ApplicationProvider.getApplicationContext(), GeoPolyActivity::class.java)
+        intent.putExtra(GeoPolyActivity.EXTRA_POLYGON, polyline)
+        val scenario = launcherRule.launch<GeoPolyActivity>(intent)
+        mapFragment.ready()
+        scenario.onActivity { activity: GeoPolyActivity ->
```
Comment (seadowg) [inline geo/src/test/java/org/odk/collect/geo/geopoly/GeoPolyActivityTest.kt:223]: All these `onActivity` calls could be combined. Same for the below test.

**Review by seadowg — CHANGES_REQUESTED** [review]

File: geo/src/test/java/org/odk/collect/geo/geopoly/GeoPolyActivityTest.kt (hunk 1 above)
Comment (grzesiek2010) [reply geo/src/test/java/org/odk/collect/geo/geopoly/GeoPolyActivityTest.kt:223]: Fixed.

**Review by seadowg — APPROVED** [review]

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on devices with Android 10, 14

Verified Cases:
- issue #6096 doesn't occur
- read only in All widgets
- regression checks in geo widgets
- Google maps, Mapbox, OSM

**Comment by WKobus** [comment]
**Tested with Success!**

Verified on device with Android 11

**Pull request merged**